packages/cli/src/utils/contentPlatform.ts

```diff
@@ -0,0 +1,312 @@
+    )
+  }
+}
+
+export function groupCustomComponentDefinitions(componentsDir: string[]) {
+  const customComponentDefinitions: Record<string, any> = {}
+  const customComponentKeys = new Set<string>()
+
+  for (const schemaFileName of componentsDir) {
+    const fileContent = readFileSync(schemaFileName, 'utf8')
+
+    try {
+      const componentSchema = JSON.parse(fileContent)
+
+      const componentKey = componentSchema.$componentKey
+
+      if (!componentKey) {
+        console.info(
+          `${chalk.red(
+            'error'
+          )} - ${schemaFileName} is missing the $componentKey property. Ignoring it.`
+        )
+
+        continue
+      }
+
+      if (customComponentKeys.has(componentKey)) {
+        console.info(
+          `${chalk.red(
+            'info'
+          )} - The component key ${componentKey} is being used more than once.\nThis can lead to unexpected behavior, please check your component schemas.\nUsing the last definition found for this key, from: ${schemaFileName}`
+        )
+      } else {
+        customComponentKeys.add(componentKey)
+      }
+
+      // Using `$componentKey` as the property name for each component in the
+      // final schema.
+      customComponentDefinitions[componentKey] = componentSchema
+    } catch (err) {
+      if (err instanceof SyntaxError) {
+        console.info(
+          `${chalk.red(
+            'error'
+          )} - ${schemaFileName} is a malformed JSON file, ignoring its content.`
+        )
+      } else {
+        throw err
+      }
+    }
+  }
+
+  return customComponentDefinitions
+}
+
+export function groupCustomContentTypeDefinitions(contentTypesDir: string[]) {
+  const customContentTypeDefinitions = {}
+  const customContentTypeKeys = new Set<string>()
+
+  for (const contentSchemaFileName of contentTypesDir) {
+    const fileContent = readFileSync(contentSchemaFileName, 'utf8')
+
+    try {
+      const contentTypeSchema = JSON.parse(fileContent)
+      const keys = Object.keys(contentTypeSchema)
+
+      for (const key of keys) {
+        if (customContentTypeKeys.has(key)) {
+          console.info(
+            `${chalk.red(
+              'info'
+            )} - The content-type ${key} is being defined more than once.\nThis can lead to unexpected behavior, please check your content-type schemas.\nUsing the last definition found, from: ${contentSchemaFileName}`
+          )
+        } else {
+          customContentTypeKeys.add(key)
+        }
+      }
+
+      Object.assign(customContentTypeDefinitions, contentTypeSchema)
+    } catch (err) {
+      if (err instanceof SyntaxError) {
+        console.info(
+          `${chalk.red(
+            'error'
+          )} - ${contentSchemaFileName} is a malformed JSON file, ignoring its content.`
+        )
+      } else {
+        throw err
+      }
+    }
+  }
+
+  return customContentTypeDefinitions
+}
+
+// This function adds a new property `$ALLOW_ALL_COMPONENTS` to the schema,
```

Comment:
question: out of curiosity, what is this allow all components?

Reply:
@lariciamota It's a shortcut for all components to be allowed on the page. Today, we have "scopes" on the hCMS to control which components can be displayed. On the Content Platform, we use JSON Schema features to control which components can be displayed. The default is an anyOf of all components available in the sections.

Reply:
So in the future this sync will be changed? Because now it's always using the `anyOf` all components, right?

Reply:
We'll always have the anyOf as default.
But the developer may want to restrict some content-types to have allowed components for a specific page.
So, the cp-sync generates anyOf lists based on the developers' schema.